### PR TITLE
support configure for minimal primary skill values

### PR DESCRIPTION
**Pull request by kdmcser**
Since I request in #4398 , it is better for modders to design minimal primary skills, so I add 4 config in game_config.json to make modder can modify them.

- "minimalAttack": 0,
- "minimalDefence": 0,
- "minimalKnowledge": 1,
- "minimalSpellPower": 1

### CODE REVIEW

File: config/gameConfig.json [hunk 1]
```diff
@@ -296,7 +296,12 @@
 			// number of artifacts that can fit in a backpack. -1 is unlimited.
 			"backpackSize"		: -1,
 			// if heroes are invitable in tavern
-			"tavernInvite"            : false
+			"tavernInvite"            : false,
+			// minimai primary skills for heroes
+			"minimalAttack": 0,
+			"minimalDefence": 0,
+			"minimalKnowledge": 1,
+			"minimalSpellPower": 1
```
Comment (IvanSavenko) [inline config/gameConfig.json:304]: No need for 4 separate values, especially if they are handled only as array
```suggestion
			"minimalPrimarySkills": [ 0, 0, 1, 1]
```

**Comment by kdmcser** [comment]
@IvanSavenko I use array and commit my code

File: lib/MimialPrimarySkill.cpp [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+};
+
+int getPrimarySkillMinimum(PrimarySkill pSkill)
+{
+    auto minialPSkills = VLC->settings()->getVector(EGameSettings::HEROES_MINIMAL_PRIMARY_SKILLS);
+    if(minialPSkills.size() != DEFAULT_MINIMAL_PSKILLS.size())
+        logGlobal->error("gameConfig.json: heroes/minimalPrimarySkills format error. need a vector with 4 elements.");
+    int index = PSKILL_INDEX_MAP.at(pSkill);
+    return minialPSkills.size() > index ? minialPSkills[index] : DEFAULT_MINIMAL_PSKILLS[index];
+}
```
Comment (IvanSavenko) [inline lib/MimialPrimarySkill.cpp:32]: All code from this file can be replaced with a single line, without any new free functions:
```cpp
VLC->settings()->getVector(EGameSettings::HEROES_MINIMAL_PRIMARY_SKILLS)[pSkill.getNum()]
```

1. PrimarySkill class has method getNum() that converts it to integer
2. We generally don't validate json via checks in code, only via schema validation (not present for game settings though, but that's separate issue)
3. We generally don't use free functions

Comment (kdmcser) [reply lib/MimialPrimarySkill.cpp]: I replace all getPrimarySkillMinimum to your suggestion, not using  free functions

**Pull request merged**